test/test_operations.py

```diff
@@ -991,6 +993,19 @@ def test_max_broadcast(self):
     xla_c = torch.max(xla_a, xla_b)
     self.assertEqual(c.data, xla_c.data.cpu())
 
+  def test_sgn(self):
+    xla_device = xm.xla_device()
+    t = torch.randn(2, 2, dtype=torch.cfloat)
+    t[0][0].real.div_(0)
```

Comment:
I think this will be `-inf`? Can we also check input to be `nan` and `inf`?

Reply:
The current scheme covers `-inf` and `nan`. I added another line to cover `inf`